app/src/main/java/org/mozilla/fenix/perf/NavGraphProvider.kt

```diff
@@ -30,12 +31,15 @@ object NavGraphProvider {
     // We want to store member state on the NavController. However, there is no way to do this.
     // Instead, we store state as part of a map: NavController instance -> State. In order to
     // garbage collect our data when the NavController is no longer relevant, we use a WeakHashMap.
-    private val map = WeakHashMap<NavController, Job>()
+    private val map = WeakHashMap<NavController, Deferred<NavGraph>>()
+    private var isGraphSet = false
```

Comment:
Assuming we need some flag to indicate the graph is set (see below for alternatives), **I don't think this is correct:** we need to have an `isGraphSet` variable for every HomeActivity/`NavController`, i.e. this needs to go in the map. I think the signature of the map should be:
```kotlin
private val map = WeakHashMap<NavController, Pair<Deferred<NavGraph>, Boolean>>()
```

Or alternatively a (data) class that represents `Pair<Deferred<NavGraph>, Boolean`.

I think you should be able to reproduce the bug by starting the app, closing it out by hitting the back button (i.e. closing HomeActivity) and relaunching the app.

Reply:
Good catch.  I'll create a private data class so each `NavController` will have a boolean tracking if the graph is set.   Thanks,